src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -34,32 +36,40 @@ class TaskKiller @Inject() (
       case Some(app) =>
         checkAuthorization(UpdateRunSpec, app)
 
+        // TODO: We probably want to pass the execution context as an implcit.
         import scala.concurrent.ExecutionContext.Implicits.global
-        taskTracker.appTasks(appId).flatMap { allTasks =>
+
+        async {
+          val allTasks = await(taskTracker.appTasks(appId))
           val foundTasks = findToKill(allTasks)
-          val expungeTasks = if (wipe) expunge(foundTasks) else Future.successful(())
-
-          expungeTasks.map { _ =>
-            val launchedTasks = foundTasks.filter(_.launched.isDefined)
-            if (launchedTasks.nonEmpty) {
-              service.killTasks(appId, launchedTasks)
-              foundTasks
-            } else foundTasks
-          }
+
+          if (wipe) expunge(foundTasks)
+
+          val launchedTasks = foundTasks.filter(_.launched.isDefined)
+          if (launchedTasks.nonEmpty) await(service.killTasks(appId, launchedTasks))
+          // TODO: For some reasons tasks returned by service.killTasks !=
+          // foundTasks in some cases. What should be returned?
```

Comment:
This change lets `TaskKillerTest "kill with wipe will kill running and expunge all"` fail. The expected return value is all killed tasks *and* all expunged tasks.